### PR TITLE
Adapt `FallbackClassifier` to graph component interface

### LINKED ISSUE #9409: Adapt `FallbackClassifier` to new `GraphComponent` interface
**Overview of the Solution**:
Adapt the `FallbackClassifier ` to the new `GraphComponent` interface.
See the [implementation proposal](https://www.notion.so/rasa/Rasa-Open-Source-3-0-Architecture-Implementation-Proposal-51ab90b05c41435ca98189a101676a1e) for further details.

* adapt `create` and `load` functions to new signatures
* implement `required_packages` and `supported_languages`
* ~adapt persistence~ (this component doesn't persist)
* ~split train and train process~ (this is not necessary for classifiers)
* support multiple messages in `process` (**make sure to handle empty messages**)

**Definition of Done**:
- [x] `FallbackClassifier` adapted to the new GraphComponent interface
- [x] Unit tests are adapted to adapted component
- [x] Other tests (integration tests, etc.) don't break (because they'd (for now) still use the old component implementation)

### PR DESCRIPTION
**Pull request by ancalita**
**Proposed changes**:
- Closes #9409 

**Status (please check what you already did)**:
- [X] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [X] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/nlu/classifiers/_fallback_classifier.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
+# flake8: noqa
+# WARNING: This module will be dropped before Rasa Open Source 3.0 is released.
+#          Please don't do any changes in this module and rather adapt
+#          FallbackClassifierGraphComponent from the regular
+#          `rasa.nlu.classifiers.sk_learn_intent_classifier` module.
```
Comment (joejuzl) [inline rasa/nlu/classifiers/_fallback_classifier.py:5]:
```suggestion
#          `rasa.nlu.classifiers.fallback_classifier.py` module.
```

File: rasa/nlu/classifiers/fallback_classifier.py [hunk 2]
```diff
@@ -17,33 +19,63 @@
...
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> None:
+        """Constructs a new fallback classifier."""
+        self.component_config = config
+        self._model_storage = model_storage
+        self._resource = resource
+        self._execution_context = execution_context
```
Comment (joejuzl) [inline rasa/nlu/classifiers/fallback_classifier.py:61]: I don't think we need any of these.

File: tests/nlu/classifiers/test_fallback_classifier.py [hunk 3]
```diff
@@ -141,17 +155,35 @@ def test_predict_fallback_intent(message: Message, component_config: Dict):
...
-    assert message == old_message_state
+    assert processed_msg == old_message_state
 
 
-def test_defaults():
-    classifier = FallbackClassifier({})
+def test_defaults(
+    default_model_storage: ModelStorage, default_execution_context: ExecutionContext,
+):
+    classifier = FallbackClassifierGraphComponent.create(
```
Comment (joejuzl) [inline tests/nlu/classifiers/test_fallback_classifier.py:181]: We could extract a helper to create the component?

**Review by joejuzl — CHANGES_REQUESTED** [review]

**Review by aeshky — APPROVED** [review]
Looks good to me.

**Review by joejuzl — APPROVED** [review]
💯

**Pull request merged**